CMakeLists.txt

```diff
@@ -168,10 +168,8 @@ function(avif_target_link_library target)
     if(NOT install_target)
         set(install_target ${target})
     endif()
-    # The transitive dependency is an export link library if it is a shared
-    # library, or if it is a static library in a static build
-    get_target_property(target_type ${target} TYPE)
-    if(target_type STREQUAL "SHARED_LIBRARY" OR NOT BUILD_SHARED_LIBS)
+    # The transitive dependency is an export link library if it is a static library in a static build.
+    if(NOT BUILD_SHARED_LIBS)
```

Comment:
I also wanted to note that we can't test this code (lines 172-174) right now, because in a static build, the cmake config files are not installed.

I tried to test this code last week but could not make it work. Another difficulty is that in a static build, the library we install is `avif_static`, not `avif`:

```
    if(BUILD_SHARED_LIBS)
        set(LIBAVIF_INSTALL_TARGET avif)
    else()
        set(LIBAVIF_INSTALL_TARGET avif_static)
    endif()
```

where `avif_static` is the merged archive library:

```
    merge_static_libs(avif_static avif)
```

I am wondering if `avif_static` does not inherit the link libraries of `avif` that we set here.

Reply:
I am playing with an installed static CI right now to debug this.